command/ca/provisioner/add.go

```diff
@@ -787,16 +800,77 @@ func createSCEPDetails(ctx *cli.Context) (*linkedca.ProvisionerDetails, error) {
+		Challenge:                     challenge,
+		Capabilities:                  ctx.StringSlice("capabilities"),
+		MinimumPublicKeyLength:        int32(ctx.Int("min-public-key-length")),
+		IncludeRoot:                   ctx.Bool("include-root"),
+		ExcludeIntermediate:           ctx.Bool("exclude-intermediate"),
+		EncryptionAlgorithmIdentifier: int32(ctx.Int("encryption-algorithm-identifier")),
+	}
+	decrypter := &linkedca.SCEPDecrypter{}
+	if decrypterCertificateFile := ctx.String("scep-decrypter-certificate-file"); decrypterCertificateFile != "" {
+		data, err := parseSCEPDecrypterCertificate(decrypterCertificateFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed parsing certificate from %q: %w", decrypterCertificateFile, err)
+		}
+		decrypter.Certificate = data
+		s.Decrypter = decrypter
+	}
+	if decrypterCertificate := ctx.String("scep-decrypter-certificate"); decrypterCertificate != "" {
+		// validate the provided value to be a valid base64 encoded PEM formatted certificate
+		data, err := base64.StdEncoding.DecodeString(decrypterCertificate)
+		if err != nil {
+			return nil, fmt.Errorf("failed base64 decoding decrypter certificate: %w", err)
+		}
+		block, rest := pem.Decode(data)
+		if len(rest) > 0 {
+			return nil, errors.New("failed parsing decrypter certificate: trailing data")
+		}
+		if block == nil {
+			return nil, errors.New("failed parsing decrypter certificate: no PEM block found")
+		}
+		if _, err := x509.ParseCertificate(block.Bytes); err != nil {
+			return nil, fmt.Errorf("failed parsing decrypter certificate: %w", err)
+		}
+		decrypter.Certificate = data
+		s.Decrypter = decrypter
+	}
+	if decrypterKeyURI := ctx.String("scep-decrypter-key-uri"); decrypterKeyURI != "" {
+		decrypter.KeyUri = decrypterKeyURI
+		s.Decrypter = decrypter
+	}
+	if decrypterKeyFile := ctx.String("scep-decrypter-key-file"); decrypterKeyFile != "" {
+		data, err := readSCEPDecrypterKey(decrypterKeyFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed reading decrypter key from %q: %w", decrypterKeyFile, err)
+		}
+		decrypter.Key = data
+		s.Decrypter = decrypter
+	}
+	if decrypterKey := ctx.String("scep-decrypter-key"); decrypterKey != "" {
+		data, err := base64.StdEncoding.DecodeString(decrypterKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed base64 decoding decrypter key: %w", err)
+		}
+		if err := validateSCEPDecrypterKey(data); err != nil {
+			return nil, fmt.Errorf("failed validating decrypter key: %w", err)
+		}
+		decrypter.Key = data
+		s.Decrypter = decrypter
+	}
+	if decrypterKeyPasswordFile := ctx.String("scep-decrypter-key-password-file"); decrypterKeyPasswordFile != "" {
+		decrypterKeyPassword, err := utils.ReadPasswordFromFile(decrypterKeyPasswordFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed reading decrypter key password from %q: %w", decrypterKeyPasswordFile, err)
+		}
+		decrypter.KeyPassword = decrypterKeyPassword
+		s.Decrypter = decrypter
```

Comment:
You can read the password first and validate the decrypter key with it.

Reply:
Thought about doing that locally in the CLI, but it's already happening on the CA side, with the somewhat handy consequence that it's not always necessary to specify the password. If the CA already knows the password, it'll be reused when the provisioner is initialized as part of validation. If there's no password, the "password required in non-interactive context" error is thrown (if not provided).